python/tk_framework_desktopserver/shotgun/api_v2.py

```diff
@@ -720,14 +719,56 @@ def _compute_sys_path(self):
     # Internal methods
 
     @sgtk.LogManager.log_timing
-    def _cache_actions(self, data, config_data):
+    def _cache_actions(self, data, config_data, cached_contents_hash=None, async=False):
         """
         Triggers the caching or recaching of engine commands.
 
         :param dict data: The data passed down from the wss client.
         :param dict config_data: A dictionary that contains, at a minimum,
             "lookup_hash", "contents_hash", "descriptor", and "entity" keys.
+        :param str cached_contents_hash: If given, represents the currently
```

Comment:
we should put a big comment in the docstring, saying that this method runs in a thread!